### ISSUE TITLE
Porous media pressure drop

### ISSUE THREAD

**Issue by rmcdermo** [firemodels/fds#3964]
Jason,
Something seems wrong with Press_2.
![screen shot 2016-06-14 at 10 29 30 am](https://cloud.githubusercontent.com/assets/4418497/16046581/f78b1de4-321a-11e6-8a1f-60fbad7ce6cb.png)

**Comment by drjfloyd** [comment]
I'll take a look.

**Comment by drjfloyd** [comment]
With the porous media the XB of the INIT is not always going to line up with the grid cells, but I want to try and capture the actual dimension of the media when it comes to computing the drag. So in each cell where we place a particle, I set the particle to dimension to only that portion of the XB that lies in the cell (e.g if the cell is 5 cm thick but only 2 cm of media is in the cell, I set the particle dimension to 2 cm).

In case 2 the INIT is not very well centered over the grid. If I take the INIT for this case and shift it slightly to be better centered, then the strange behavior goes away. Not sure what this points to in terms of any solution.

**Comment by rmcdermo** [comment]
Kevin,
Please have a look.

**Comment by mcgratta** [comment]
OK, I'll look at this tomorrow.

**Comment by mcgratta** [comment]
I'm not sure this is a problem. The case demonstrates that the model is being implemented correctly. In case 2, there's just something that triggers the flow out of its smooth, laminar state, and that affects the pressure drop a bit.

**Comment by rmcdermo** [comment]
Well, then something is wrong with the expected value for that case.  So,
it may not be a bug, but it IS a problem in terms of the verification case.

**Comment by drjfloyd** [comment]
The expected value is correct. As I stated in my previous post, if I make a
slight shift in the INIT (keep the same x span but move it so it is evenly
split over grid cells), it remains smooth and has the same value as the
other two cases.

**Comment by mcgratta** [comment]
I made some progress on this case. The slight shift that Jason mentions fixes things because of the way we extrapolate momentum transfer from particle to grid. Currently, we assign all of a particle's momentum (`LP%ACCEL_X`, for example) to the closest cell face. In the medium resolution case above, three layers of cell-centered particles represent the porous slab, two of which transfer their momentum to the right, and one to the left. The one that shifts to the left just happens to be at 0.498 of the cell width because of the less than precise look-up table we use to get a particle's cell index. The real issue is that cell-centered particles throw their weight to the left or right arbitrarily. I changed this to linearly weight the particle's momentum onto the two adjacent cell faces. For these screens and porous media, this is going to smooth the pressure drop a bit and it's less likely to stir up wiggles. I still think the wiggles are going to appear as the volume fraction of the porous media goes to zero, but that's another issue.

I have not pushed up any code. I need to see where we're at with the latest release. I want to test this more.
